targets/TARGET_STM/README.md

```diff
@@ -25,3 +68,235 @@ This table summarizes the STM32Cube versions currently used :
+
+But also each STM32 Part Number with different FLASH size : STM32F401xC / STM32F401xE
+
+MBED porting layer specific for this family are placed here.
+
+Example in TARGET_STM32G0:
+- TARGET_STM32G031xx
+- TARGET_STM32G071xx
+- ...
+
+Each STM32 sub-family contains:
+- toolchains files
+- board specific files
+
+
+### Add a custom board
+
+ST provides the complete support for few NUCLEO and DISCO boards.
+
+Locate one of these boards with the minimum difference with your chosen MCU.
+
+Copy paste, and update!
+
+
+### Board specific files (pinmap)
+
+2 files in mbed:
+- PeripheralPins.c
+- PinNames.h
+
+Tip is to use a python script to generate those files
+
+https://github.com/ARMmbed/mbed-os/blob/master/tools/targets/STM32_gen_PeripheralPins.py
+
+STM32CubeMX has to be installed first. Path has to be specified in the `cube_path.json` file.
+
+```
+$ python tools/targets/STM32_gen_PeripheralPins.py -h
+
+Script version 1.9
+usage: STM32_gen_PeripheralPins.py [-h] [-l | -b | -m xml | -t HW | -c CUSTOM]
+
+Script will generate PeripheralPins.c thanks to the xml files description available in
+STM32CubeMX directory defined in 'cube_path.json':
+        C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -l, --list            list available mcu xml files description in STM32CubeMX
+  -b, --boards          list available boards description in STM32CubeMX
+  -m xml, --mcu xml     specify the mcu xml file description in STM32CubeMX to use (use double quotes).
+                           Parameter can be a filter like L496 if you want to parse all L496 chips (-m STM32 to parse all).
+  -t HW, --target HW    specify the board file description in STM32CubeMX to use (use double quotes).
+                           Parameter can be a filter like L496 (only the first file found will be parsed).
+  -c CUSTOM, --custom CUSTOM
+                        specify a custom board .ioc file description to use (use double quotes).
+
+Once generated, you have to check and comment pins that can not be used (specific HW, internal ADC channels, remove PWM using us ticker timer, ...)
+```
+
+How to generate files for a custom boards based on a STM32F427 MCU:
+```
+$ python tools/targets/STM32_gen_PeripheralPins.py -l | grep F427
+STM32F427A(G-I)Hx.xml
+STM32F427I(G-I)Hx.xml
+STM32F427I(G-I)Tx.xml
+STM32F427V(G-I)Tx.xml
+STM32F427Z(G-I)Tx.xml
+
+$ python tools/targets/STM32_gen_PeripheralPins.py -m "STM32F427V(G-I)Tx.xml"
+
+Script version 1.9
+CubeMX DB version DB.5.0.60
+
+ * Output directory: C:\github\mbed\STM32F427V(G-I)Tx
+ * Generating PeripheralPins.c and PinNames.h with 'STM32F427V(G-I)Tx.xml'
+ * GPIO file: C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeMX\db\mcu\IP\GPIO-STM32F427_gpio_v1_0_Modes.xml
+ * I/O pins found: 82 connected: 0
+```
+
+### Use of custom_targets.json
+
+https://os.mbed.com/docs/mbed-os/v6.0-preview/porting/porting-a-custom-board.html
```

Comment:
We'll need to use this link for the release
https://os.mbed.com/docs/mbed-os/v6.0/porting/porting-a-custom-board.html

Reply:
done